packages/core/src/sessionManagement.ts

```diff
@@ -89,25 +123,50 @@ function retrieveSession(sessionCookie: CookieCache): SessionState {
   return session
 }
 
-export function persistSession(session: SessionState, cookie: CookieCache) {
+export function persistSession(session: SessionState, cookie: CookieCache, withNewSessionStrategy = false) {
+  if (withNewSessionStrategy && !utils.isEmptyObject(session)) {
+    session.expire = String(Date.now() + SESSION_EXPIRATION_DELAY)
+  }
   const cookieString = utils
     .objectEntries(session)
     .map(([key, value]) => `${key}=${value}`)
     .join(SESSION_ENTRY_SEPARATOR)
-  cookie.set(cookieString, EXPIRATION_DELAY)
+  cookie.set(cookieString, SESSION_EXPIRATION_DELAY)
 }
 
 export function stopSessionManagement() {
-  registeredActivityListeners.forEach((e) => e())
-  registeredActivityListeners = []
+  registeredListeners.forEach((e) => e())
+  registeredIntervals.forEach((interval) => clearInterval(interval))
+  registeredListeners = []
+  registeredIntervals = []
 }
 
-let registeredActivityListeners: Array<() => void> = []
+let registeredListeners: Array<() => void> = []
+let registeredIntervals: number[] = []
 
 export function trackActivity(expandOrRenewSession: () => void) {
+  const doExpandOrRenewSession = monitor(expandOrRenewSession)
   const options = { capture: true, passive: true }
-  ;['click', 'touchstart', 'keydown', 'scroll'].forEach((event: string) => {
-    document.addEventListener(event, expandOrRenewSession, options)
-    registeredActivityListeners.push(() => document.removeEventListener(event, expandOrRenewSession, options))
+  ;[utils.DOM_EVENT.CLICK, utils.DOM_EVENT.TOUCH_START, utils.DOM_EVENT.KEY_DOWN, utils.DOM_EVENT.SCROLL].forEach(
+    (event: string) => {
+      document.addEventListener(event, doExpandOrRenewSession, options)
+      registeredListeners.push(() => document.removeEventListener(event, doExpandOrRenewSession, options))
+    }
+  )
+}
+
+function trackVisibility(expandSession: () => void, visibilityStateProvider: () => VisibilityState) {
```

Comment:
IMO
* `trackVisibility` should expose an `Observable` instead of a callback function
* be self contained: no `visibilityStateProvider`, use `document.visibilityState` directly. We can mock `document.visibilityState` in tests

Reply:
> trackVisibility should expose an Observable instead of a callback function

what advantage do you see with it?

> We can mock document.visibilityState in tests

How would you do that?

Reply:
> what advantage do you see with it?

I think having a common strategy to pass data streams around would make things more future-proof. For now, our Observable implementation is quite limited, but in the future it could be extendend with a "destructor" when unused ([example](https://github.com/zenparsing/zen-observable/blob/master/README.md#new-observablesubscribe)), and we would have a common way to remove listeners and stuff. This would make functions like this more self-contained and easier to test.

But if you feel that this is thinking too far ahead, I'm okay to keep a callback for now :)

> How would you do that?

```js
Object.defineProperty(document, 'visibilityState', { get() { return "zog" }, configurable: true })
// then
delete document.visibilityState
```

Reply:
About observable instead of callbacks, I am not sure that observable pattern should be enforce in every situation.
I think it:
- allow us to easily reuse a data source in another part of the code
- can help us to decouple some of the browser API to the domain logic 
But it also adds an extra layer of abstraction.
About future-proof arguments, let's see when we will have the need to do more complicated stuff with our observable and address it at this time.

In this case, I am not sure to see much benefit to switch to observable.